### ISSUE TITLE
Odd representation of leading .. in import paths

### ISSUE THREAD

**Issue by quasicomputational** [dhall-lang/dhall-haskell#406]
To represent an import path like `../foo/bar` using the API, you need to write `Local Parent (File (Directory ["foo"]) "bar")`. But importing `../../foo/bar` needs `Local Parent (File (Directory ["foo", ".."]) "bar")`; the first `..` gets special treatment. And importing `./../../foo/bar`, which is equivalent, could be represented as `Local Here (File (Directory ["foo", "..", ".."]) "bar")`.

I think it would be simpler and more uniform to just drop the `Parent` constructor from `FilePrefix`, and put `..` segments in the `Directory`. Alternatively, if it's desired to keep `..` out of the `Directory`, then the `Here` constructor could take a `Natural` saying how many directories to go upwards, and then the documentation will need to state that `..` segments shouldn't be present in a `Directory`.

**Comment by Gabriel439** [comment]
@quasicomputational: Yeah, that seems reasonable.  The only reason the Haskell implementation does it this way is to match the standard as closely as possible, but what you're proposing is semantically equivalent
